src/main/java/org/jboss/pnc/environmentdriver/Driver.java

```diff
@@ -284,6 +288,27 @@ public CompletionStage<EnvironmentCreateResponse> create(
         });
     }
 
+    private void processPod(Pod pod, boolean sidecarEnabled, boolean sidecarArchiveEnabled) {
+        if (!sidecarEnabled) {
+            List<Container> containers = pod.getSpec().getContainers();
+            Optional<Container> sidecarContainer = containers.stream()
+                    .filter(c -> "sidecar".equals(c.getName()))
+                    .findFirst();
+            if (!sidecarContainer.isEmpty()) {
+                containers.remove(sidecarContainer.get());
+            }
+        }
+        if (!sidecarArchiveEnabled) {
+            List<Container> initContainers = pod.getSpec().getInitContainers();
+            Optional<Container> sidecarInitContainer = initContainers.stream()
+                    .filter(c -> "sidecar-init".equals(c.getName()))
```

Comment:
We should document "sidecar-init" and "sidecar" in the pod definition being special, and will be removed if the sidecar and sidecarArchiveEnabled are false.

Perhaps both in the readme and in the API definition for environment-driver.

Reply:
+1 for the changes to the documentation, we already agreed with @pkocandr to put comments inside the pod template definition to make it clear that those names are important and should be updated (if ever) together with the Driver code. But ok to document anywhere sensible.

Wrt to the new configuration property, I have written an improvement change in https://issues.redhat.com/browse/NCL-7692. It would be less confusing to have the properties inside a single place IMO. I can take it if you want, don't want to add more work on you.

Reply:
The README.md has been updated.